### ISSUE TITLE
Git commands don't work on RHEL 6

### ISSUE THREAD

**Issue by weshaggard** [dotnet/sourcelink#197]
```
/root/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63201-01/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.DllNotFoundException: Unable to load shared library '/root/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63201-01/build/../tools/netcoreapp2.0/runtimes/rhel-x64/native/libgit2-b0d9952.so' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by /root/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63201-01/build/../tools/netcoreapp2.0/runtimes/rhel-x64/native/libgit2-b0d9952.so) [/git/corefx/external/netstandard/netstandard.depproj]

/root/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63201-01/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.InternalLoadUnmanagedDllFromPath(String unmanagedDllPath) [/git/corefx/external/netstandard/netstandard.depproj]
/root/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63201-01/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.LoadUnmanagedDllFromPath(String unmanagedDllPath) [/git/corefx/external/netstandard/netstandard.depproj]
/root/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63201-01/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.GitLoaderContext.LoadUnmanagedDll(String unmanagedDllName) in /_/src/Microsoft.Build.Tasks.Git/GitLoaderContext.cs:line 48 [/git/corefx/external/netstandard/netstandard.depproj]
/root/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63201-01/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.ResolveUnmanagedDll(String unmanagedDllName, IntPtr gchManagedAssemblyLoadContext) [/git/corefx/external/netstandard/netstandard.depproj]
/root/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63201-01/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_libgit2_init() [/git/corefx/external/netstandard/netstandard.depproj]
/root/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63201-01/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.InitializeNativeLibrary()
[/git/corefx/external/netstandard/netstandard.depproj]
/root/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63201-01/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods..cctor() [/git/corefx/external/netstandard/netstandard.depproj]
/root/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63201-01/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    --- End of inner exception stack trace --- [/git/corefx/external/netstandard/netstandard.depproj]
/root/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63201-01/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_buf_free(GitBuf buf) [/git/corefx/external/netstandard/netstandard.depproj]
/root/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63201-01/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.Proxy.ConvertPath(Func`2 pathRetriever)
[/git/corefx/external/netstandard/netstandard.depproj]
/root/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63201-01/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Repository.Discover(String startingPath) [/git/corefx/external/netstandard/netstandard.depproj]
/root/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63201-01/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.RepositoryTasks.LocateRepository(LocateRepository task) in /_/src/Microsoft.Build.Tasks.Git.Operations/RepositoryTasks.cs:line 50 [/git/corefx/external/netstandard/netstandard.depproj]
/root/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63201-01/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.LocateRepository.Execute() in /_/src/Microsoft.Build.Tasks.Git/LocateRepository.cs:line 22 [/git/corefx/external/netstandard/netstandard.depproj]
/root/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63201-01/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/git/corefx/external/netstandard/netstandard.depproj]
/root/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63201-01/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/git/corefx/external/netstandard/netstandard.depproj]
```

For now corefx will just disable SourceLink on this platform but I thought I'd file an issue to consider fixing this for platforms that don't have GLIBC.

**Comment by bording** [comment]
@tmat LibGit2Sharp 0.26.0-preview-0062 has some changes to the `rhel-x64 binaries that I believe should fix this problem.

**Comment by danmosemsft** [comment]
We seem to be periodically broken by this in the .NET Core repos. Here is one tracking issue:
https://github.com/dotnet/corefx/issues/33595 
Always a TypeLoadException in LibGit2Sharp.Core.NativeMethods.
It looks like it may be always Ubuntu 18.04. Linked issue suggests it did not like the libc it found, as above.

**Comment by danmosemsft** [comment]
@tmat did Source Link pick up the updated LibGit2Sharp mentioned above?

**Comment by tmat** [comment]
Not yet. I hope to look into it next week.

**Comment by ViktorHofer** [comment]
Please prioritize this right as it currently blocks local Ubuntu 18.04 and WSL builds in corefx.

**Comment by danmosemsft** [comment]
@ViktorHofer which WSL distro is this? Was it Ubuntu 1804?

**Comment by ViktorHofer** [comment]
vihofer@DESKTOP-JB1LJPE:~$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.1 LTS
Release:        18.04
Codename:       bionic

**Comment by et1975** [comment]
It also fails on OpenSUSE, **even when my project is not under git (has no `.git` folder)**:
```
~/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018: The "Microsoft.Build.Tasks.Git.LocateRepository" task failed unexpectedly. [snip]
~/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.PlatformNotSupportedException: opensuse-leap.15.0-x64 [snip]
~/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.RuntimeIdMap.GetNativeLibraryDirectoryName(String runtimeIdentifier) in /_/src/Microsoft.Build.Tasks.Git/RuntimeIdMap.cs:line 26 [snip]
~/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.GitLoaderContext.GetNativeLibraryDirectory() in /_/src/Microsoft.Build.Tasks.Git/GitLoaderContext.cs:line 56 [snip]
~/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.GitLoaderContext.LoadUnmanagedDll(String unmanagedDllName) in /_/src/Microsoft.Build.Tasks.Git/GitLoaderContext.cs:line 31 [snip]
~/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.ResolveUnmanagedDll(String unmanagedDllName, IntPtr gchManagedAssemblyLoadContext) [snip]
~/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_libgit2_init() [snip]
~/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.InitializeNativeLibrary() [snip]
~/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods..cctor() [snip]
~/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    --- End of inner exception stack trace --- [snip]
~/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_buf_free(GitBuf buf) [snip]
~/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.Proxy.ConvertPath(Func`2 pathRetriever) [snip]
~/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Repository.Discover(String startingPath) [snip]
~/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.RepositoryTasks.LocateRepository(LocateRepository task) in /_/src/Microsoft.Build.Tasks.Git.Operations/RepositoryTasks.cs:line 50 [snip]
~/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.LocateRepository.Execute() in /_/src/Microsoft.Build.Tasks.Git/LocateRepository.cs:line 22 [snip]
~/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [snip]
~/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [snip]

Build FAILED.
```

Please make git targets opt-in.

**Comment by omajid** [comment]
FWIW, I am getting this error when running ./build.sh in corefx on a fresh Fedora 25 machine too.

```
/home/ci/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63401-01/build/Microsoft.Build.Tasks.Git.targets(20,5): warning : The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. [/home/ci/corefx/src/Native/build-native.proj]
/home/ci/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63401-01/build/Microsoft.Build.Tasks.Git.targets(20,5): warning :    at LibGit2Sharp.Core.NativeMethods.git_buf_free(GitBuf buf) [/home/ci/corefx/src/Native/build-native.proj]       
/home/ci/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63401-01/build/Microsoft.Build.Tasks.Git.targets(20,5): warning :    at LibGit2Sharp.Core.Proxy.ConvertPath(Func`2 pathRetriever) [/home/ci/corefx/src/Native/build-native.proj]      
/home/ci/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63401-01/build/Microsoft.Build.Tasks.Git.targets(20,5): warning :    at LibGit2Sharp.Repository.Discover(String startingPath) [/home/ci/corefx/src/Native/build-native.proj]
/home/ci/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63401-01/build/Microsoft.Build.Tasks.Git.targets(20,5): warning :    at Microsoft.Build.Tasks.Git.GitOperations.LocateRepository(String directory) in /_/src/Microsoft.Build.Tasks.Git.Operations/GitOperations.cs:line 26 [/home/ci/corefx/src/Native/build-native.proj]                                       
/home/ci/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63401-01/build/Microsoft.Build.Tasks.Git.targets(20,5): warning :    at Microsoft.Build.Tasks.Git.RepositoryTasks.LocateRepository(LocateRepository task) in /_/src/Microsoft.Build.Tasks.Git.Operations/RepositoryTasks.cs:line 58 [/home/ci/corefx/src/Native/build-native.proj]                             
/home/ci/.nuget/packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.18613.4/tools/Version.targets(69,5): error : SourceRevisionId is not set, which means the SourceLink targets are not included in the build. Those are needed to produce a correct sha for our build outputs. [/home/ci/corefx/src/Native/build-native.proj]
```

**Comment by tmat** [comment]
Should be fixed in version `1.0.0-beta2-18618-05`.
